packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -842,6 +853,13 @@ DataGridPremiumRaw.propTypes = {
    * @default false
    */
   pagination: PropTypes.bool,
+  /**
+   * The extra information about the pagination state of the Data Grid.
+   * Only applicable with `paginationMode="server"`.
+   */
+  paginationMeta: PropTypes.shape({
+    hasNextPage: PropTypes.bool,
```

Comment:
Another interface could be to have `props.hasNextPage` callback, the reason I chose a meta object:
1. I am not exactly sure what params should be passed to this callback, paginationModel? (would be the same as onPaginationModelChange), rows? (could easily be fetched?), what else?
1. When we support cursor pagination internally in the data source, we'll have to store more values to it, like nextCursor, previousCursor, hasPreviousPage, etc.
1. All these values seem to be interrelated and should be invoked at the same times, for example, if hasNextPage updates, it's likely that nextCursor also updates (and also the others). So to avoid splitting them now and combining them later like we did with (page, pageSize) -> paginationModel, it seems more reasonable to have a structure like this from the start.

Opinions are welcome

Reply:
If it is only for `paginationMode="server"`, you could consider giving it a name that provides this information (like `serverPaginationMeta`)

Reply:
That could be better, yes. Though I wanted to make `paginationMeta` a second argument of the `onPaginationModelChange` call so that if some cursor information is stored there, the user can access it without using an additional selector, for that purpose I was more inclined not to make the naming more generic and not bound to the server-side. I am not sure yet, but I have a feeling there might be more general use cases where it could be useful. 🤷